src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1672,7 +1672,8 @@ private CorInfoType getFieldType(CORINFO_FIELD_STRUCT_* field, ref CORINFO_CLASS
             TypeDesc fieldType = fieldDesc.FieldType;
             CorInfoType type = asCorInfoType(fieldType, out structType);
 
-            Debug.Assert(_compilation.TypeSystemContext.GetWellKnownType(WellKnownType.ByReferenceOfT).GetKnownField("_value").FieldType.Category == TypeFlags.IntPtr);
+            Debug.Assert(_compilation.TypeSystemContext.GetWellKnownType(WellKnownType.ByReferenceOfT, false) == null ||
```

Comment:
This is relatively expensive to execute for every getFieldType even before this change. Change it to:

`Debug.Assert(fieldDesc.OwningType.IsByReferenceOfT && ... the long condition here ...)`

?